### PR TITLE
feat(recordings): show 'out of band' events in recordings

### PR DESCRIPTION
**Pull request by rcmarron**
## Problem

Non `posthog-js` events are excluded from the events shown in a recording. This is because those events are not technically a part of the recording (not tagged with the correct window and session ids).

However, users want to see these events in the recording. Particularly in the case of segment events or backend events. (came in from the user slack)

## Changes

The implementation here is not obvious because a recording as a user sees can actually be multiple recordings spliced together (e.g. flipping back and forth between two different tabs). If an event does not originate from a specific tab, then we don't know which tab it should be shown with.

The approach taken here takes a best guess at which tab to show the event with. It leans on the server's generation of 'recording segments' which accounts for user activity when determining which tab to show the user.

There are also some UX changes to clarify these events to the user.

1) Out of band events have muted text (see $identify)

image

2) There's a tooltip explaining the 'out of band' events and that the timing might not be precise

image

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Added test cases for out of band events and manually verified that backend events appeared where expected.

### CODE REVIEW

**Comment by rcmarron** [comment]
@clarkus I took a stab at doing the UI here on my own, but very happy to change it up if you have suggestions. I think there's a bigger pass we should on this events list when we want to tackle 'sessions without recordings', but I was hoping to do a more surgical fix here.

Also, of note, the way it's build right now, we can't have rows with different heights in this list (because virtualization + autoscroll), so we have some limitations there.

**Comment by clarkus** [comment]
I think this works for now, but longer term I think we should focus more on differentiating this as a type. It feels like some alternate icon style (something less prominent / opaque, but isolated to the icon so the rest of the event is legible) and probably communicating the source of the event if possible. It really feels like an extension of taxonomy but in the context of recordings. I think the tooltip is great as it's clear and obvious.

File: frontend/src/scenes/session-recordings/player/playerUtils.ts [hunk 1]
```diff
@@ -109,11 +109,33 @@ export function getPlayerPositionFromPlayerTime(
     return null
 }
 
+// Gets the player position from an epoch-time without a window-id
+// Used to place backend events in a recording even though, they don't
+// have a window id
+export function guessPlayerPositionFromEpochTimeWithoutWindowId(
```
Comment (EDsCODE) [inline frontend/src/scenes/session-recordings/player/playerUtils.ts:115]: lmao nice

File: frontend/src/scenes/session-recordings/player/playerUtils.ts [hunk 2]
```diff
@@ -109,11 +109,33 @@ export function getPlayerPositionFromPlayerTime(
...
+    }
+    return null
+}
+
 export function getPlayerPositionFromEpochTime(
     epochTime: number,
     windowId: string,
     startAndEndTimesByWindowId: Record<string, RecordingStartAndEndTime>
 ): PlayerPosition | null {
+    if (!windowId && !startAndEndTimesByWindowId['']) {
```
Comment (EDsCODE) [inline frontend/src/scenes/session-recordings/player/playerUtils.ts:133]: What's supposed to happen here?

Comment (rcmarron) [reply frontend/src/scenes/session-recordings/player/playerUtils.ts]: Gahh, good catch. That wasn't supposed to be there... this case is handled in the `sessionRecordingLogic`. I had briefly considered putting the logic in the util, but decided against it.

**Comment by rcmarron** [comment]
> I think this works for now, but longer term I think we should focus more on differentiating this as a type. It feels like some alternate icon style (something less prominent / opaque, but isolated to the icon so the rest of the event is legible) and probably communicating the source of the event if possible.

100% agree.

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**